### PR TITLE
Ref #7873 and #7874. RTOS concepts

### PR DESCRIPTION
**Pull request by pablogs9**
This PR completes the required RTOS information

Please merge when this is merged:
- [x] https://github.com/micro-ROS/micro-ROS.github.io/pull/121

### CODE REVIEW

File: _docs/concepts/rtos/index.md [hunk 1]
```diff
@@ -9,24 +9,44 @@ Bare-metal applications are also used nowadays, but it requires a very low-level
...
+
+<br/><br/>
+
+### Zephyr RTOS
+
+<br/><br/>
+
+<img align="left" width="140" style="padding-right: 25px;"  src="https://upload.wikimedia.org/wikipedia/commons/2/2d/Zephyr-logo.png">
+
+[Zephyr](https://www.zephyrproject.org/) RTOS is based on a small-footprint kernel designed for use on resource-constrained and embedded systems: from simple embedded environmental sensors and LED wearables to sophisticated embedded controllers, smart watches, and IoT wireless applications. The Zephyr kernel supports multiple architectures, including ARM Cortex-M, Intel x86, ARC, NIOS II, Tensilica Xtensa and RISC-V 32. Zephyr implements a subset of the embedded [POSIX profiles](https://docs.zephyrproject.org/latest/guides/portability/posix.html). In order to obtain more information, please visit [Zephyr getting started guide](https://docs.zephyrproject.org/latest/introduction/index.html).
```
Comment (julibert) [inline _docs/concepts/rtos/index.md]:
```suggestion
[Zephyr](https://www.zephyrproject.org/) RTOS is based on a small-footprint kernel designed for use on resource-constrained and embedded systems: from simple embedded environmental sensors and LED wearables to sophisticated embedded controllers, smartwatches, and IoT wireless applications. The Zephyr kernel supports multiple architectures, including ARM Cortex-M, Intel x86, ARC, NIOS II, Tensilica Xtensa and RISC-V 32. Zephyr implements a subset of the embedded [POSIX profiles](https://docs.zephyrproject.org/latest/guides/portability/posix.html). In order to obtain more information, please visit [Zephyr getting started guide](https://docs.zephyrproject.org/latest/introduction/index.html).
```

File: _docs/concepts/rtos_comparison/index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+* NuttX
+* FreeRTOS
+* Zephyr
+
+regarding the following features:
+* Standardized API to Application level
+* Maturity
+* Supported Hardware (regarding Olimex, Bosch XDK, MPC57xx)
+* Scheduling options
+* IO Support (native or vendor specific module needed)
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
* IO Support (native or vendor-specific module needed)
```

File: _docs/concepts/rtos_comparison/index.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+* Memory Footprint
+* Safety Certification
+* License
+
+Further to investigate:
+* POSIX level support
+* RIOT RTOS in general
+
+Key questions:
+* Evaluation POSIX-complience of RTOS.
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
* Evaluation POSIX-compliance of RTOS. 
```

File: _docs/concepts/rtos_comparison/index.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+| Semaphore /Mutex Management                                  | yes (Priority Inheritance)                                          | yes                                                                                                                      | yes                                                                                                      |
+| **IO**                                                       |                                                                     |                                                                                                                          |                                                                                                          |
+| I2C                                                          | yes                                                                 | vendor specific                                                                                                          | yes                                                                                                      |
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
| I2C                                                          | yes                                                                 | vendor-specific                                                                                                          | yes                                                                                                      |
```

File: _docs/concepts/rtos_comparison/index.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+| SPI                                                          | yes                                                                 | vendor specific                                                                                                          | yes                                                                                                      |
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
| SPI                                                          | yes                                                                 | vendor-specific                                                                                                          | yes                                                                                                      |
```

File: _docs/concepts/rtos_comparison/index.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+| UART                                                         | hw-specific                                                         | vendor specific                                                                                                          | yes                                                                                                      |
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
| UART                                                         | hw-specific                                                         | vendor-specific                                                                                                          | yes                                                                                                      |
```

File: _docs/concepts/rtos_comparison/index.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+| USB                                                          | yes                                                                 | vendor specific                                                                                                          | yes                                                                                                      |
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
| USB                                                          | yes                                                                 | vendor-specific                                                                                                          | yes                                                                                                      |
```

File: _docs/concepts/rtos_comparison/index.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+| CAN                                                          | yes                                                                 | vendor specific                                                                                                          | yes                                                                                                      |
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
| CAN                                                          | yes                                                                 | vendor-specific                                                                                                          | yes                                                                                                      |
```

File: _docs/concepts/rtos_comparison/index.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+| CAnopen                                                      | no                                                                  | vendor specific                                                                                                          | yes                                                                                                      |
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
| CAnopen                                                      | no                                                                  | vendor-specific                                                                                                          | yes                                                                                                      |
```

File: _docs/concepts/rtos_comparison/index.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+| Modbus                                                       | yes                                                                 | vendor specific                                                                                                          | ?                                                                                                        |
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
| Modbus                                                       | yes                                                                 | vendor-specific                                                                                                          | ?                                                                                                        |
```

File: _docs/concepts/rtos_comparison/index.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+<sup>1</sup> Processes, signals, fpe, segmentation, bus errors, timers, file and directory ops, pipes, c library, IO Port Interface
+
+<sup>2</sup> Real-time, clocks, semaphores, messages, shared mem, async io, memory locking.
+
+<sup>3</sup> Threads.
+
+<sup>4</sup> Executing every task in round-robin fashion but only for a pre-defined time slice.
+
+<sup>5</sup> ARM Cortex M3 EFM32GG390F1024 Giant Gecko family (Silicon Labs).[EFM32G880F120-STK](http://nuttx.org/doku.php?id=documentation:about)ARM Cortex M3 EFM32GG390F1024 Giant Gecko family (Silicon Labs).
```
Comment (julibert) [inline _docs/concepts/rtos_comparison/index.md]:
```suggestion
<sup>5</sup> ARM Cortex M3 EFM32GG390F1024 Giant Gecko family (Silicon Labs). [EFM32G880F120-STK](http://nuttx.org/doku.php?id=documentation:about) ARM Cortex M3 EFM32GG390F1024 Giant Gecko family (Silicon Labs).
```

File: _docs/concepts/rtos_comparison/index.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+| **Scheduling**                                               |                                                                     |                                                                                                                          |                                                                                                          |
+| Priority-based                                               | FIFO                                                                | yes                                                                                                                      | yes                                                                                                      |
+| Round-Robin <sup>4</sup>                                     | yes                                                                 | yes <sup>6</sup>                                                                                                         | [co-operative](http://docs.zephyrproject.org/kernel/threads/scheduling.html)                             |
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply _docs/concepts/rtos_comparison/index.md]: Done

**Comment by ralph-lange** [comment]
The comparison should be put now at: https://micro-ros.github.io/docs/concepts/rtos/comparison/

**Comment by pablogs9** [comment]
Ok, let me refactor this PR and I'll request you for a review

File: _docs/concepts/rtos/comparison/index.md [hunk 13]
```diff
@@ -3,4 +3,107 @@ title: Comparison of these RTOS
...
+
+<sup>6</sup> [Note: Time slicing](https://www.freertos.org/Documentation/161204_Mastering_the_FreeRTOS_Real_Time_Kernel-A_Hands-On_Tutorial_Guide.pdf)
+
+<sup>7</sup> Hardware-support for Networking and Storage often depends on the platform and sometimes packages of hardware-vendors are available, which work for a particular operating system. But it is in general difficult to determine the harware-support of a given RTOS.
+
+Some Related Work:
+* [Choosing the right RTOS for IoT platform, Milinkovic et al, INFOTEH-JAHORINA Vol. 14, 2015](http://infoteh.rs.ba/zbornik/2015/radovi/RSS-2/RSS-2-2.pdf): comparison of FreeRTOS, ChibiOS/RT, Erika, RIOT
+* [FreeRTOS Architecture](http://www.aosabook.org/en/freertos.html)
+
+## Supported development boards
```
Comment (ralph-lange) [inline _docs/concepts/rtos/comparison/index.md]: The following two paragraphs can be found at: https://micro-ros.github.io/docs/concepts/rtos/NuttX/

I think they can be removed here.

Comment (pablogs9) [reply _docs/concepts/rtos/comparison/index.md]: Done

File: _docs/concepts/rtos/comparison/index.md [hunk 14]
```diff
@@ -3,4 +3,99 @@ title: Comparison of these RTOS
...
+
+<sup>5</sup> ARM Cortex M3 EFM32GG390F1024 Giant Gecko family (Silicon Labs). EFM32G880F120-STK ARM Cortex M3 EFM32GG390F1024 Giant Gecko family (Silicon Labs).
+
+<sup>6</sup> [Note: Time slicing](https://www.freertos.org/Documentation/161204_Mastering_the_FreeRTOS_Real_Time_Kernel-A_Hands-On_Tutorial_Guide.pdf)
+
+<sup>7</sup> Hardware-support for Networking and Storage often depends on the platform and sometimes packages of hardware-vendors are available, which work for a particular operating system. But it is in general difficult to determine the harware-support of a given RTOS.
+
+Some Related Work:
+* [Choosing the right RTOS for IoT platform, Milinkovic et al, INFOTEH-JAHORINA Vol. 14, 2015](http://infoteh.rs.ba/zbornik/2015/radovi/RSS-2/RSS-2-2.pdf): comparison of FreeRTOS, ChibiOS/RT, Erika, RIOT
+* [FreeRTOS Architecture](http://www.aosabook.org/en/freertos.html)
```
Comment (ralph-lange) [inline _docs/concepts/rtos/comparison/index.md]: I propose to add this link also to the FreeRTOS page.

Comment (pablogs9) [reply _docs/concepts/rtos/comparison/index.md]: Done

File: _docs/overview/index.md [hunk 15]
```diff
@@ -20,7 +20,7 @@ Major repositories in order of the layers are:
 
 * Applications:
   * Kobuki demo: [embedded robot driver](https://github.com/micro-ROS/apps/tree/kobuki_rcl_port/examples/kobuki) and [remote ROS 2 software](https://github.com/micro-ROS/micro-ROS_kobuki_demo)
-  * [Temperature demo](https://github.com/micro-ROS/micro-ROS_temperature_publisher_demo)
+  * Broken URL: Temperature demo
```
Comment (ralph-lange) [inline _docs/overview/index.md]:
```suggestion
```

Let's remove it completely for now.

**Pull request merged**